test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/Expressions/FilterBinderTests.cs

```diff
@@ -2956,6 +2956,25 @@ public void FilterByDynamicProperty()
                 "$it => (Convert(IIF((($it.ProductProperties != null) AndAlso $it.ProductProperties.ContainsKey(Token)), $it.ProductPropertiesToken, null)) == \"1\")");
         }
 
+        [Fact]
+        public void InOnPrimitiveCollectionPropertyOnRHS()
+        {
+            var filters = VerifyQueryDeserialization(
+               "42 in AlternateIDs",
+               "$it => $it.AlternateIDs.Contains(42)",
+               NotTesting);
+
+            RunFilters(
+                filters,
+                new Product { AlternateIDs = new[] { 1, 2, 42 } },
+                new { WithNullPropagation = true, WithoutNullPropagation = true });
+
+            RunFilters(
+                filters,
+                new Product { AlternateIDs = new[] { 1, 2 } },
```

Comment:
The original reported issue seemed to imply this was an issue with `string` typed collection. Does it matter if we test that?

Reply:
It wouldn't make any difference really. The issue was reproduceable with any array of primitives - `string[]`, int[]`, `decimal[]`, `double[]`, etc. The root of the problem was the introduction of code in [this](https://github.com/OData/WebApi/pull/2195/files#diff-0f1b878d20db0edc22bbe7cb23057c8ea3e3dffad198ea5044920fff31515b42R522) PR that treated everything like a generic collection.